src/github.com/getlantern/flashlight/client/pluggable_transports.go

```diff
@@ -86,6 +90,19 @@ func obfs4DialFactory(s *ChainedServerInfo, deviceID string) (dialFN, error) {
 	}
 
 	return func() (net.Conn, error) {
+		ctxProxyInfo(s.Addr, "obfs4")
 		return cf.Dial("tcp", s.Addr, net.Dial, args)
 	}, nil
 }
+
+func ctxProxyInfo(addr string, protocol string) {
+	m := ctx.Map{
+		"proxy_protocol": protocol,
+	}
+	host, port, err := net.SplitHostPort(addr)
+	if err == nil {
+		m["proxy_host"] = host
+		m["proxy_port"] = port
+	}
+	ctx.SetAll(m)
```

Comment:
One of the interesting things that came out of this exercise is that I didn't need global helper functions.  Since the context only needs to be set in the place where it's know, that logic can just be encapsulated there.  This keeps the ctx package itself very generic and uncoupled from the specifics of flashlight/lantern.